### PR TITLE
Add option_value_fields metadata 

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
The `OptionValue` table has a number of fields which are not relevant to every `OptionGroup`, such as "description", "icon" and "color". This adds `option_value_fields` metadata about each `OptionGroup`, which was previously hard-coded in various forms.

Before
----------------------------------------
No metadata to tell what `OptionValue` fields are relevant to which `OptionGroup`.

After
----------------------------------------
Metadata exists, and is returned by APIv4 GetFields, so it's now possible to tell which field options support a "color" or an "icon", etc.

### CODE REVIEW

File: CRM/Admin/Form/Options.php [hunk 1]
```diff
@@ -174,11 +178,12 @@ public function buildQuickForm() {
...
     }
-    else {
+
+    // Add icon & color if this option group supports it.
+    if ($optionGroup['shape'] && in_array('icon', $optionGroup['shape'])) {
       $this->add('text', 'icon', ts('Icon'), ['class' => 'crm-icon-picker', 'title' => ts('Choose Icon'), 'allowClear' => TRUE]);
     }
-
-    if (in_array($this->_gName, ['activity_status', 'case_status'])) {
+    if ($optionGroup['shape'] && in_array('color', $optionGroup['shape'])) {
```
Comment (colemanw) [inline CRM/Admin/Form/Options.php:186]: This is the upshot of the change for our forms, we no longer need to hard-code a list of `OptionGroup`s that support "color" or "icon" we can rely on the metadata.

**Comment by eileenmcnaughton** [comment]
I tested this & the field is correctly added / populated on upgrade and can be accessed via the api.

When adding an activity type I was offered the icon (but not colour) option before and after

My one reservation is the field name 'shape' wasn't meaningful to me - and we will be living with it for a while - @seamuslee001 @totten how do you feel about the field name 'shape'?

**Comment by seamuslee001** [comment]
I wonder if properties might be a better name or fields maybe?

**Comment by colemanw** [comment]
I got the idea from PHPStorm which uses [Array Shapes](https://blog.jetbrains.com/phpstorm/2022/02/phpstorm-2022-1-eap-3/), (which are very cool by the way!) where the "shape" describes what keys an array is expected to have. It's a slightly odd term but I still can't think of a better one.

**Comment by eileenmcnaughton** [comment]
@colemanw array shapes do look pretty cool - they differ a bit from this in that this is just a flat list of fields supported for the option group whereas array shapes describes an array whose 'shape' could vary siginifcantly (depth, string types)

I like them though

**Comment by colemanw** [comment]
I don't want this PR to stall on a bike-sheddy thing. I'm willing to change the name to something clearer. How about `option_value_fields`? That's about as literal as you can get to describe this column's purpose.

**Comment by eileenmcnaughton** [comment]
I'm OK with that -

**Comment by seamuslee001** [comment]
Sounds fine to me

**Comment by colemanw** [comment]
Merging per tag & agreement about the name.

**Comment by eileenmcnaughton** [comment]
@colemanw thanks - I was going to give it one more r-run - but didn't see any issues with the code

**Pull request merged**